core/shared/src/main/scala/scalaz/zio/Duration.scala

```diff
@@ -0,0 +1,73 @@
+package scalaz.zio
+
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.{ Duration => ScalaDuration, FiniteDuration => ScalaFiniteDuration }
+import scala.math.Ordered
+
+sealed trait Duration extends Ordered[Duration] {
+
+  def +(other: Duration): Duration
+
+  def *(factor: Double): Duration
+
+  def max(other: Duration): Duration = if (this > other) this else other
+
+  def min(other: Duration): Duration = if (this < other) this else other
+
+}
+
+final object Duration {
+
+  object Finite {
+
+    def apply(nanos: Long): Duration =
+      if (nanos >= 0) new Finite(nanos)
+      else Infinity
+
+  }
+
+  final case class Finite private (nanos: Long) extends Duration {
+
+    def +(other: Duration): Duration = other match {
+      case Finite(otherNanos) => new Finite(nanos + otherNanos)
+      case Infinity           => Infinity
+    }
+
+    def *(factor: Double): Duration =
+      if (!factor.isInfinite && !factor.isNaN) new Finite((nanos * factor).round)
```

Comment:
I think that this can lead to problems if `factor < 0`.

This method should either check that, or call `Finite.apply` instead of using `new Finite`.

Reply:
good catch